### PR TITLE
Feature/batch dev

### PR DESCRIPTION
**Pull request by tjsego**
Updating with all current batch execution functionality.

### CODE REVIEW

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py [hunk 1]
```diff
@@ -641,18 +707,115 @@ def load_rendering_manipulator(self, gd_manipulator, trial_idx=0, mcs=0):
...
+            self.gd.field_extractor.setSimulationData(sim_data_int_addr)
+
+            for field_name, screenshot_data in self.scm.screenshotDataDict.items():
+                min_max = self._get_field_min_max(screenshot_data)
+                if min_max is not None:
+                    if field_name not in min_max_dict.keys():
+                        min_max_dict[field_name] = min_max
+                    else:
+                        if min_max[0] < min_max_dict[field_name][0]:
+                            min_max_dict[field_name][0] = min_max[0]
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py:757]: Why not

`min_max_dict[field_name][0] = min(min_max_dict[field_name][0], min_max[0])`

?

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py [hunk 2]
```diff
@@ -696,4 +862,331 @@ def render_trial_results(self, trial_idx):
...
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py:1065]: same, why not min(a,b)

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/batch_run.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+                        mults[x] = float(mults[x])
+                    except ValueError:
+                        raise ValueError(f'Multiplier {mults[x]} for {var} is not a number')
+
+        check_multipliers(mult_dict)
+
+    assert num_rep > 0, 'Number of replicas per parameter set (num_rep) must be greater than zero'
+    if num_par < 1:
+        num_par = 1
+    assert out_freq >= 0, 'Output frequency (out_freq) must be greater than zero'
```
Comment (JulianoGianlupi) [inline CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/batch_run.py]: String should say greater or equal

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py (hunk 1 above)
Comment (tjsego) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py:757]: Either works

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py (hunk 2 above)
Comment (tjsego) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py:1065]: Same. Either works.

**Review by JulianoGianlupi — APPROVED** [review]

File: CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/batch_run.py (hunk 3 above)
Comment (tjsego) [reply CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/batch_run.py]: Fixed in 37897b3

**Pull request merged**